tests/solvers/test_base_solver.py

```diff
@@ -0,0 +1,67 @@
+from typing import Tuple
+
+import pytest
+
+import numpy as np
+import scipy.sparse as sp
+
+from moscot.base.output import MatrixSolverOutput
+from tests._utils import ATOL, RTOL, MockSolverOutput
+
+
+class TestBaseSolverOutput:
+    @pytest.mark.parametrize("batch_size", [1, 4])
+    @pytest.mark.parametrize("threshold", [0.0, 1e-1, 1.0])
+    @pytest.mark.parametrize("shape", [(7, 2), (91, 103)])
+    def test_sparsify_threshold(self, batch_size: int, threshold: float, shape: Tuple[int, int]) -> None:
+        rng = np.random.RandomState(42)
+        tmap = np.abs(rng.rand(shape[0], shape[1]))
+        output = MockSolverOutput(tmap)
+        mso = output.sparsify(mode="threshold", value=threshold, batch_size=batch_size)
+        assert isinstance(mso, MatrixSolverOutput)
```

Comment:
This check is not redundant, as I was explaining, since you're accessing it's attributes later (and if it weren't of the correct type, it would fail).